internal/entitlement/postgresadapter/entitlement.go

```diff
@@ -159,6 +188,10 @@ func mapEntitlementEntity(e *db.Entitlement) *entitlement.Entitlement {
 	}
 
 	if e.UsagePeriodAnchor != nil && e.UsagePeriodInterval != nil {
+		ent.GenericProperties.UsagePeriod = &entitlement.UsagePeriod{
+			Anchor:   *convert.SafeToUTC(e.UsagePeriodAnchor),
```

Comment:
```suggestion
			Anchor:   e.UsagePeriodAnchor.In(time.UTC)
```

nit: given that you already done the nil check, I wouldn't involve all the dereferring magic here.